### ISSUE TITLE
Issue with hasSpecial pattern validator

### ISSUE THREAD

**Issue by tmeloliveira** [anandts88/ember-validator#8]
I'm trying to use the `pattern` validator to validate a password. My code looks like this:

``` javascript
  pattern: {
    hasNumber: true,
    hasSpecial: '!@#$&*',
    messages: {
      hasNumber: 'Password should contain at least one number',
      hasSpecial: 'Password should have at least special'
    }
  },
```

But somehow the it was not working. So went debugging in your add-on's code, and found an issue.

I'm using version `1.2.7`, and the issue was on the file `ember-validator/addon/mixins/pattern.js`. 

In this part of the code, at the end of every regex you are doing the test with `.test(value)`.

``` javascript
if (this.options.hasSpecial) {
  if (typeof(this.options.hasSpecial) === 'string') {
    specialTest = new RegExp('(?=.*[' + this.options.hasSpecial + '])').test(value);
  } else if (this.options.hasSpecial.constructor !== RegExp) {
    specialTest = this.options.hasSpecial.test(value);
  } else {
    specialTest = new RegExp('(?=.*[!@$%^&*()-+_=~`{}:;"\'<>,.|?])').test(value);
  }
}
```

So this is making the variable `specialTest` a boolean, right ? Because of `.test(value)` result.

But later on in the code, you are doing this test:

``` javascript
} else if (this.options.hasSpecial && !specialTest.test(value)) {
    this.pushResult(this.options.messages.hasSpecial);
}
```

And that's causing an error, because `!specialTest.test(value)` is trying to call the method `test` in a boolean variable, and not in a regex.

I think the best solution would be to remove the `.test(value)` from where you define `specialTest`.

I hope it helps! :)

**Comment by anandts88** [comment]
@tmeloliveira Thanks. This comment is really valuable and will address the issue as soon I can. I will reply back to this thread once it is resolved. Your suggestions are always helps us to bring up a better product.

**Comment by tmeloliveira** [comment]
@anandts88 thanks for the fast reply as well! :) I'm happy to contribute in your project.
Looking forward to the next release to update my dependencies! Hehehe!

**Comment by tmeloliveira** [comment]
Hey @anandts88! I don't mean to bother you but... Do you know when is `1.3.2` coming out ? I'm holding a bugfix waiting for it... 

Is there any other task pending ? Perhaps I could help! :)

**Comment by anandts88** [comment]
@tmeloliveira As part of 1.3.2 we are releasing updates for doing computed validations (performing validations when user is typing over the field). We are almost done and working on README updates related to this. 1.3.2 will be live on next Monday Mar 7th. Apologize for the delay.

**Comment by tmeloliveira** [comment]
No problem! Looking forward to it! :)

**Comment by anandts88** [comment]
@tmeloliveira We published 1.3.2. Please validate and close this issue.

**Comment by tmeloliveira** [comment]
It's working! Thank you! :)
